src/ExtEventLoop.php

```diff
@@ -25,13 +25,31 @@ class ExtEventLoop implements LoopInterface
     private $readListeners = [];
     private $writeListeners = [];
     private $running;
+    private $signals;
+    private $signalEvents = [];
 
     public function __construct(EventBaseConfig $config = null)
     {
         $this->eventBase = new EventBase($config);
         $this->futureTickQueue = new FutureTickQueue();
         $this->timerEvents = new SplObjectStorage();
 
+        $this->signals = new SignalsHandler();
+        $this->signals->on('on', function ($signal) {
+            $this->signalEvents[$signal] = Event::signal($this->eventBase, $signal, function () use ($signal) {
+                $this->signals->call($signal);
+            });
+            $this->signalEvents[$signal]->add();
+        });
+        $this->signals->on('off', function ($signal) {
+            $this->futureTick(function () use ($signal) {
```

Comment:
Does this have to be in a future tick? This looks unexpected from a consumer perspective and would at least need some documentation here? I haven't verified this, but it looks like this could potentially cause a race condition, can you verify this?